flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -331,6 +331,22 @@
                                     + "JobManager could be faster, since no reverse DNS lookup is performed. "
                                     + "However, local input split assignment (such as for HDFS files) may be impacted.");
 
+    @Documentation.Section(Documentation.Sections.EXPERT_JOB_MANAGER)
+    public static final ConfigOption<Integer> JOB_MANAGER_FUTURE_THREADS =
+            key("jobmanager.future-executor.numThreads")
+                    .intType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "The number of jobmanager future-executor threads for shared services.");
```

Comment:
I think we should add that this defaults to the number of CPUs if no value has been specified.

Reply:
`JobManagerOptions` lives within `flink-core` so I don't think we can access `Hardware` in `flink-runtime` so I specify the default value as recommended earlier with `config.getInteger(JobManagerOptions.JOB_MANAGER_FUTURE_THREADS, Hardware.getNumberCPUCores())`.

Reply:
Yes, this is correct. I think we can mention the default in the description.

Reply:
Sounds good!